jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/ContentDemander_state.puml

```diff
@@ -0,0 +1,25 @@
+@startuml
+
+null:
+content:
+DEMANDING:
+EOF:
+
+[*] --> null
+
+null --> DEMANDING : demand()
+null --> EOF : eof()
```

Comment:
what about `onContent`, `onTimeout` etc?

Reply:
`onContent` cannot be called when you're not in `DEMANDING` state. `onTimeout` is a no-op in the cases I did not include, but I've added them to be complete.